File: leetcode_hot100/283 移动零/code.ts
/**
 Do not return anything, modify nums in-place instead.
 */
 function moveZeroes(nums: number[]): void {
    let NoneZeroPtr:number = 0;
    for(let i:number = 0; i < nums.length; i++){
        if (nums[i] !== 0){
            nums[NoneZeroPtr] = nums[i];
            NoneZeroPtr++;
        }
    }
    for(let i:number = NoneZeroPtr; i < nums.length; i++){
        nums[i] = 0
    }
};

File: leetcode_hot100/双指针/283 移动零/code.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        NoneZeroPtr = 0
        for i in range(0, len(nums)):
            if (nums[i] != 0):
                nums[NoneZeroPtr] = nums[i]
                NoneZeroPtr += 1
        for i in range(NoneZeroPtr, len(nums)):
            nums[i] = 0